src/Catalyst.Abstractions/Kvm/IDeltaResolver.cs

```diff
@@ -21,19 +21,16 @@
 
 #endregion
 
-using Catalyst.Protocol.Deltas;
-using Nethermind.Core.Crypto;
+using Lib.P2P;
 
 namespace Catalyst.Abstractions.Kvm
 {
     public interface IDeltaResolver
     {
-        Keccak Resolve(int deltaNumber);
+        bool TryResolve(long deltaNumber, out Cid deltaHash);
 
-        Delta Latest { get; }
+        long LatestDeltaNumber { get; }
```

Comment:
Should this be int rather than long? Would we have a negative Δ?

@Xela101 in your sync work what data type are you using for delta number for consistency.

Reply:
I'm not receiving the delta number yet, I will when I merge this into my code. I think long is probably better because of the size or ulong since the delta number should never be negative and gives 2x the size on the positive.

Reply:
👍